ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,146 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+import six
+import abc
+import re
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RedHatAuthTool(object):
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthTool):
+
+    def __get_authselect_current_output(self):
+        current = ipautil.run(
+            [
+                paths.AUTHSELECT,
+                "current"
+            ],
+            env={"LC_ALL": "C.UTF8"}
+        )
+
+        if current.returncode != 0:
+            return None
+
+        return current.raw_output.decode()
+
+    def __parse_authselect_output(self, output_text=None):
+        if output_text is None:
+            output_text = self.__get_authselect_current_output()
+
+        if output_text is None:
```

Comment:
Remove the extra empty line and move the second ``if output_text is None`` inside the first if block.

Reply:
ok. but what will it change?

Reply:
It makes the code cleaner and your intent more obvious.